src-built-in/components/advancedAppLauncher/src/stores/StoreActions.js

```diff
@@ -144,6 +145,30 @@ function appInAppList(appName) {
 	let app = findAppByField('name', appName);
 	return Boolean(app);
 }
+
+/**
+ * Ensures all 'apps' properties on folders conform
+ * to the new structure (Array vs object)
+ */
+function validateFolderDataStructure() {
+	Object.keys(data.folders).map(folderName => {
+		const folder = data.folders[folderName];
+		if (!Array.isArray(folder.apps)) {
+			const warning = "Advanced App Launcher distributed store foundation appears to use incorrect format. Please ensure all folders[folderName].apps are arrays instead of objects";
+			// If the structure is wrong, notify the user in hopes that the foundation will be fixed
+			FSBL.Clients.Logger.warn(warning);
+			FSBL.UserNotification.alert("system", "ONCE-SINCE-STARTUP", "Distributed Store Type Mismatch", warning);
```

Comment:
That notification needs better wording. Maybe - Application Launcher Persistent Store has data stored in deprecated format - or something like that.